packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  VCIssuerVerification,
+  ClaimIssuerVerification,
+  CredentialResolver,
+  IssuerResolver,
+  OffChainClaim,
+} from '.';
+import { VerifiableCredential } from '@ew-did-registry/credentials-interface';
+import type { RoleCredentialSubject } from '@energyweb/credential-governance';
+import { NoCredential } from './errors';
+import { addressOf } from '@ew-did-registry/did-ethr-resolver';
+
+/**
+ * A class to provide verification of issuer authority for either VC or OffChainClaim
+ */
+export class IssuerVerification {
+  private credentialResolver: CredentialResolver;
+  private vcIssuerVerification: VCIssuerVerification;
+  private claimIssuerverification: ClaimIssuerVerification;
+
+  constructor(
+    private issuerResolver: IssuerResolver,
+    credentialResolver: CredentialResolver,
+    vcIssuerVerification: VCIssuerVerification,
+    claimIssuerverification: ClaimIssuerVerification
+  ) {
+    this.credentialResolver = credentialResolver;
+    this.vcIssuerVerification = vcIssuerVerification;
+    this.claimIssuerverification = claimIssuerverification;
+  }
+
+  /**
+   * Verifies issuer authority with either Verifiable Credential or OffChainClaim
+   *
+   * ```typescript
+   * const issuerVerification = new IssuerVerification(
+   * credentialResolver,
+   * vcIssuerVerification,
+   * claimIssuerVerification,
+   * );
+   *
+   * let issuer : 'did:ethr:volta:0x...';
+   * const role = 'role';
+   * await issuerVerification.verifyIssuer(issuer, role);
+   * ```
+   *
+   * @param issuer issuer DID to verify authority
+   * @param role authoritative role to be issuer
+   * @returns
+   */
+  async verifyIssuer(issuer: string, role: string) {
```

Comment:
Could you implement this similarly to https://github.com/energywebfoundation/ew-credentials/blob/55b84926b85a7e2261fffad8231821736d5febb6/packages/vc-verification/src/vc-issuer-verification.ts#L38? It would be more readable

Reply:
Do you mean the error handling? Or the call structure?

Reply:
I think of them would be beneficial